src/lib/sedlexing.ml

```diff
@@ -369,62 +495,6 @@ module Utf8 = struct
             lor (n4 land 0x3f)
         | _ -> assert false
 
-    let from_gen s =
-      let next_or_fail () =
-        match Gen.next s with None -> raise MalFormed | Some x -> Char.code x
-      in
-      Gen.next s >>| fun c1 ->
-      match width c1 with
-        | 1 -> Uchar.of_char c1
-        | 2 ->
-            let n1 = Char.code c1 in
-            let n2 = next_or_fail () in
-            if n2 lsr 6 != 0b10 then raise MalFormed;
-            Uchar.of_int (((n1 land 0x1f) lsl 6) lor (n2 land 0x3f))
-        | 3 ->
-            let n1 = Char.code c1 in
-            let n2 = next_or_fail () in
-            let n3 = next_or_fail () in
-            if n2 lsr 6 != 0b10 || n3 lsr 6 != 0b10 then raise MalFormed;
-            Uchar.of_int
-              (((n1 land 0x0f) lsl 12)
-              lor ((n2 land 0x3f) lsl 6)
-              lor (n3 land 0x3f))
-        | 4 ->
-            let n1 = Char.code c1 in
-            let n2 = next_or_fail () in
-            let n3 = next_or_fail () in
-            let n4 = next_or_fail () in
-            if n2 lsr 6 != 0b10 || n3 lsr 6 != 0b10 || n4 lsr 6 != 0b10 then
-              raise MalFormed;
-            Uchar.of_int
-              (((n1 land 0x07) lsl 18)
-              lor ((n2 land 0x3f) lsl 12)
-              lor ((n3 land 0x3f) lsl 6)
-              lor (n4 land 0x3f))
-        | _ -> raise MalFormed
```

Comment:
I'm not sure where that logic is is in the new code.

Also, I believe these unicode-specific functions should be placed in a helper module outside of the generator code so they can be tested in isolation for strict adherence to the unicode standard..

Reply:
This is no longer used, it was mostly a duplication of the function `next`, we now use that function `next` in all code path,